lib/trento/application/process_managers/deregistration_process_manager.ex

```diff
@@ -44,22 +49,47 @@ defmodule Trento.DeregistrationProcessManager do
   """
   def interested?(%HostRegistered{host_id: host_id}), do: {:start, host_id}
   def interested?(%HostRolledUp{host_id: host_id}), do: {:start, host_id}
+  def interested?(%HostAddedToCluster{host_id: host_id}), do: {:start, host_id}
+  def interested?(%ClusterRolledUp{snapshot: %{hosts: hosts}}), do: {:start, hosts}
+
   def interested?(%HostDeregistrationRequested{host_id: host_id}), do: {:continue, host_id}
+
   def interested?(%HostDeregistered{host_id: host_id}), do: {:stop, host_id}
+
```

Comment:
Shouldn't the code listen to `HostRemovedFromCluster` to run the `apply` and remove the cluster from the state?
Otherwise, the process manager is ignorant of this process and would run a non needed `DeregisterClusterHost`. Keep in mind that, the commands could be launched somewhere else

Reply:
Good catch!